plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/Connection.java

```diff
@@ -88,6 +90,23 @@ public Connection(IClient client, ICredentialsPrompter credentialsPrompter, ISSL
 		this.sslCertificateCallback = sslCertCallback;
 	}
 	
+	@Override
+	public Properties getExtendedProperties() {
+		return extendedProperties;
+	}
+
+	@Override
+	public void setExtendedProperty(String name, String value) {
+		Properties oldExt = new Properties(extendedProperties);
```

Comment:
my point is if your properties contain foo=bar, and you call setExtendedProperty("foo","bar"), then it'll fire a change event, but it shouldn't